concordium-std/src/impls.rs

```diff
@@ -2699,6 +2865,66 @@ impl Deserial for HashKeccak256 {
     }
 }
 
+impl Serial for ExchangeRates {
+    fn serial<W: Write>(&self, out: &mut W) -> Result<(), W::Err> {
+        self.euro_per_energy.serial(out)?;
+        self.micro_ccd_per_euro.serial(out)?;
+        Ok(())
+    }
+}
+
+impl Deserial for ExchangeRates {
+    fn deserial<R: Read>(source: &mut R) -> ParseResult<Self> {
+        let bytes: [u8; 32] = source.read_array()?;
+        let chunks = unsafe { mem::transmute::<[u8; 32], [[u8; 8]; 4]>(bytes) };
+
+        let euro_per_energy_numerator = u64::from_le_bytes(chunks[0]);
+        let euro_per_energy_denominator = u64::from_le_bytes(chunks[1]);
+        let micro_ccd_per_euro_numerator = u64::from_le_bytes(chunks[2]);
+        let micro_ccd_per_euro_denominator = u64::from_le_bytes(chunks[3]);
+
+        if euro_per_energy_numerator == 0
+            || euro_per_energy_denominator == 0
+            || micro_ccd_per_euro_numerator == 0
+            || micro_ccd_per_euro_denominator == 0
+        {
+            return Err(ParseError::default());
+        }
+
+        let euro_per_energy =
+            ExchangeRate::new_unchecked(euro_per_energy_numerator, euro_per_energy_denominator);
+        let micro_ccd_per_euro = ExchangeRate::new_unchecked(
+            micro_ccd_per_euro_numerator,
+            micro_ccd_per_euro_denominator,
+        );
+
+        Ok(Self {
+            euro_per_energy,
+            micro_ccd_per_euro,
+        })
+    }
+}
+
+impl Serial for AccountBalance {
+    fn serial<W: Write>(&self, out: &mut W) -> Result<(), W::Err> {
+        self.total.serial(out)?;
+        self.staked.serial(out)?;
+        self.locked.serial(&mut *out)?;
```

Comment:
Why is the &mut *out reborrow here needed?

Reply:
Ohh... not sure why that was inserted